ui/components-react/src/components-react/propertygrid/internal/flat-items/MutableCategorizedArrayProperty.ts

```diff
@@ -50,7 +53,18 @@ export class MutableCategorizedArrayProperty
     const childrenDepth = depth + (this._renderLabel ? 1 : 0);
     this._children = record.getChildrenRecords().map((child, index) => {
       const newName = `${child.property.name}_${index}`;
-      const newDisplayLabel = `[${index + 1}] ${child.property.displayLabel}`;
+      let newDisplayLabel = `[${index + 1}]`;
+
+      if (
+        child.description &&
+        ((child.value.valueFormat === PropertyValueFormat.Array &&
+          child.value.items.length > 1) ||
+          (child.value.valueFormat === PropertyValueFormat.Struct &&
+            Object.keys(child.value.members).length > 1))
+      ) {
```

Comment:
why was this condition necessary?

Reply:
Because labels for primitive values are taken from the description, so there would be duplicate labels for primitive array items like this:
[1] Description:    Description

Reply:
And if I recall correctly, structs and arrays that have one element are flattened to display that element instead?

Reply:
> Because labels for primitive values are taken from the description, so there would be duplicate labels for primitive array items like this:
> [1] Description: Description

Yes, and I think that's the only check we should do - if the array item is not primitive.

> And if I recall correctly, structs and arrays that have one element are flattened to display that element instead?

That happens before the property data gets to `@itwin/components-react`. Once it's here, no more modifications are made to it.

Reply:
Gotcha, removed extra conditions